projects/js-packages/charts/src/components/pie-semi-circle-chart/pie-semi-circle-chart.tsx

```diff
@@ -24,66 +24,77 @@ interface PieSemiCircleChartProps extends BaseChartProps< DataPointPercentage[]
 	 */
 	thickness?: number;
 
+	/**
+	 * Padding around the chart
+	 */
+	padding?: number;
+
+	/**
+	 * Direction of chart rendering
+	 * true for clockwise, false for counter-clockwise
+	 */
+	clockwise?: boolean;
+
 	/**
 	 * Label text to display above the chart
 	 */
 	label?: string;
+
 	/**
 	 * Note text to display below the label
 	 */
 	note?: string;
+
 	/**
-	 * Direction of chart rendering
-	 * true for clockwise, false for counter-clockwise
+	 * Additional class name for the chart container
 	 */
-	clockwise?: boolean;
+	className?: string;
 }
 
 type ArcData = PieArcDatum< DataPointPercentage >;
 
+/**
+ * Validates the semi-circle pie chart data
+ * @param data - The data to validate
+ * @return Object containing validation result and error message
+ */
+const validateData = ( data: DataPointPercentage[] ) => {
+	if ( ! data.length ) {
+		return { isValid: false, message: 'No data available' };
+	}
+
+	// Check for negative values
+	const hasNegativeValues = data.some( item => item.percentage < 0 || item.value < 0 );
+	if ( hasNegativeValues ) {
+		return { isValid: false, message: 'Invalid data: Negative values are not allowed' };
+	}
+
+	// Validate total percentage is greater than 0
+	const totalPercentage = data.reduce( ( sum, item ) => sum + item.percentage, 0 );
+	if ( totalPercentage <= 0 ) {
+		return { isValid: false, message: 'Invalid percentage total: Must be greater than 0' };
+	}
+
+	return { isValid: true, message: '' };
+};
+
 const PieSemiCircleChart: FC< PieSemiCircleChartProps > = ( {
 	data,
-	width,
+	width = 400,
+	thickness = 0.4,
+	padding = 20,
```

Comment:
I would recommend not to add new features (it introduced `padding` and `className`) for a PR that only suppose to introduce tests and fixes if any. Would that make sense?

Reply:
yeah, that makes sense. in the testing it was revealed that this chart, although it's based upon the pie chart, is missing some of the pie chart functionality. But I do agree that this could be better in its own PR : )